src/main/java/org/breedinginsight/brapi/v2/services/BrAPITrialService.java

```diff
@@ -52,4 +93,278 @@ private long countGermplasm(UUID programId, String trialDbId) throws ApiExceptio
         List<BrAPIObservationUnit> obUnits = ouDAO.getObservationUnitsForTrialDbId(programId, trialDbId);
         return obUnits.stream().map(BrAPIObservationUnit::getGermplasmDbId).distinct().count();
     }
+
+    public DownloadFile exportObservations(
+            Program program,
+            UUID experimentId,
+            ExperimentExportQuery params) throws IOException, DoesNotExistException, ApiException {
+        StreamedFile downloadFile;
+        boolean isDataset = false;
+        List<BrAPIObservation> dataset = new ArrayList<>();
+        List<BrAPIObservationVariable> obsVars = new ArrayList<>();
+        Map<String, Map<String, Object>> rowByOUId = new HashMap<>();
+        Map<String, BrAPIStudy> studyByDbId = new HashMap<>();
+        List<String> requestedEnvNames = StringUtils.isNotBlank(params.getEnvironments()) ?
```

Comment:
It may be better for this code to assume that `environments` is a list of environment UUIDs instead of names.  That can help the code further down in this method to be more succinct

Reply:
agreed. done

Reply:
agreed, and made the change